bindings/python/scripts/clang_utils.py

```diff
@@ -0,0 +1,114 @@
+    """
+
+    results = []
+    names = dir(object)
+    # :dd any DynamicClassAttributes to the list of names if object is a class;
+    # this may result in duplicate entries if, for example, a virtual
+    # attribute with the same name as a DynamicClassAttribute exists
+    try:
+        base_members = filter(
+            lambda k, v: isinstance(v, types.DynamicClassAttribute),
+            object.__bases__.__dict__.items(),
+        )
+        names.extend(base_members)
+    except AttributeError:
+        pass
+    for key in names:
+        value = inspect.getattr_static(object, key)
+        if not predicate or predicate(value):
+            results.append((key, value))
+    results.sort(key=lambda pair: pair[0])
+    return results
+
+
+class ClangUtils:
+    """
+    Clang's cindex class utilities.
+
+    Supports the following objects:
+        CursorKind:
+            - A CursorKind describes the kind of entity that a cursor points to.
+            - https://github.com/llvm/llvm-project/blob/1acd9a1a29ac30044ecefb6613485d5d168f66ca/clang/bindings/python/clang/cindex.py#L657
+        Cursor:
+            - The Cursor class represents a reference to an element within the AST. It acts as a kind of iterator.
+            - https://github.com/llvm/llvm-project/blob/1acd9a1a29ac30044ecefb6613485d5d168f66ca/clang/bindings/python/clang/cindex.py#L1415
+        Type:
+            - The Type class represents the type of an element in the abstract syntax tree.
+            - https://github.com/llvm/llvm-project/blob/1acd9a1a29ac30044ecefb6613485d5d168f66ca/clang/bindings/python/clang/cindex.py#L2180
+    """
+
+    def __init__(self, object):
+        if not (
+            isinstance(object, clang.CursorKind)
+            or isinstance(object, clang.Cursor)
+            or isinstance(object, clang.Type)
+        ):
+            raise NotImplementedError(f"Not implemented for {object}")
+
+        self.check_functions_dict = {}
+        self.get_functions_dict = {}
+        self.properties_dict = {}
+
+        # A list to ignore the functions/properties that causes segmentation errors.
+        ignore_list = [
+            "mangled_name",
+            "get_address_space",
+            "get_typedef_name",
+            "tls_kind",
+        ]
+
+        # populate dicts
+        for entry in getmembers_static(object):
+            if entry[0] not in ignore_list:
+                if inspect.isfunction(entry[1]):  # if function
+                    try:
+                        if entry[0].startswith("is_"):
+                            self.check_functions_dict[entry[0]] = entry[1](object)
+                        if entry[0].startswith("get_"):
+                            self.get_functions_dict[entry[0]] = entry[1](object)
+                    except:
+                        continue
+                else:  # else, property
+                    try:
```

Comment:
`isinstance` doesn't throw. Why is this try here?

Reply:
`getattr` throws error.
Can change to:
```diff
-                    try:
-                        if isinstance(entry[1], property):
+                    if isinstance(entry[1], property):
+                        try:
                             self.properties_dict[entry[0]] = getattr(object, entry[0])
-                    except:
-                        continue
+                        except:
+                            continue
```